src/data/resolvers/transactions.ts

```diff
@@ -211,3 +213,41 @@ export const getColonyUnclaimedTransfers = async (
 
   return transfers.filter(notUndefined);
 };
+
+export const transactionResolvers = ({
+  colonyManager: { networkClient },
+}: Required<Context>): Resolvers => ({
+  Query: {
+    async transaction(_, { transactionHash, colonyAddress }) {
+      const { provider } = networkClient;
+      const colonyClient = await networkClient.getColonyClient(colonyAddress);
+      /*
+       * @TODO Optimze! Optimize! Optimize!
+       * We need to find a way to fetch specific events related to transaction
+       * As the way it's currently implement is not scalable at all...
+       */
+      const colonyEvents = await getColonyAllEvents(colonyClient);
```

Comment:
I think I can improve this currently. Just try to fetch the event & log for the current transaction hash.

If that's is one of our contract tx's then it's an "action", otherwise it's just a normal transaction.

Then we can deal later with fetching subsequent events for this one particular transaction. But at any rate, it's better than fetching the whole events list and filtering it.

Reply:
Nope. I was wrong.

Turns out that currently we don't have a way to fetch this for a single transaction hash, so for now we still have to rely on fetching all then filtering.

This is something I have to figure out, the most likely have to build it in `colonyJS`

But as it is, it can stay like this for now.

Reply:
Nope again. I was wrong about being wrong.

Turns out Alex saved me once again. :tada: 

I'll put it in.